### PR TITLE
Set fact fixes

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME

modules/set_fact

### CODE REVIEW

File: lib/ansible/plugins/action/set_fact.py [hunk 1]
```diff
@@ -37,25 +38,29 @@ def run(self, tmp=None, task_vars=None):
...
                 if not isidentifier(k):
-                    result['failed'] = True
-                    result['msg'] = ("The variable name '%s' is not valid. Variables must start with a letter or underscore character, and contain only "
-                                     "letters, numbers and underscores." % k)
-                    return result
+                    raise AnsibleActionFail("The variable name '%s' is not valid. Variables must start with a letter or underscore character, "
+                                            "and contain only letters, numbers and underscores." % k)
 
+                #NOTE: this should really use BOOLEANS from convert_bool, but only in the k=v case, right now it converts matching explicit yaml strings also.
                 if not C.DEFAULT_JINJA2_NATIVE and isinstance(v, string_types) and v.lower() in ('true', 'false', 'yes', 'no'):
```
Comment (s-hertel) [inline lib/ansible/plugins/action/set_fact.py]: Doesn't checking DEFAULT_JINJA2_NATIVE first address that though?

File: changelogs/fragments/fix_set_fact.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - Corrected and expanded documentation as well as now raise errors that were previouslly ignored.
```
Comment (s-hertel) [inline changelogs/fragments/fix_set_fact.yml]: s/previouslly/previously/

Could prefix with `set_fact -`

File: lib/ansible/plugins/action/set_fact.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/plugins/action/set_fact.py]: only when native, when 'non native' it still makes these 2 the same:
```
normallybool: no
normallystring: "no"
```

Comment (s-hertel) [reply lib/ansible/plugins/action/set_fact.py]: when non native, it seems like they both become booleans anyway

Comment (bcoca) [reply lib/ansible/plugins/action/set_fact.py]: yes, that is my point, they really should not, but not changing implementation at this point, why i added note, clarified a bit more now

Comment (s-hertel) [reply lib/ansible/plugins/action/set_fact.py]: Oh, I misread your comment too (took "right now" to mean "in the case of using convert_bool" as opposed to devel)

**Pull request merged**